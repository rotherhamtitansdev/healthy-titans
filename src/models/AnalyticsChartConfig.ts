import { NHSNutritionalDataModel } from "./NHSNutritionalDataModel";

// This is the numerical instance of the NHS Nutritional guidelines model
export const NHSNutritionalBoundaries: NHSNutritionalDataModel[] = [
  { name: "Fat", low: 3, high: 17.5 },
  { name: "Saturates", low: 1.5, high: 5 },
  { name: "Sugars", low: 5, high: 22.5 },
  { name: "Salt", low: 0.1, high: 0.6 },
];

// These are the config parameters for the chart styling and sizing
export const chartSizing = {
  chartHeight: 500,
  barWidth: 30,
  CStartPos: 50,
  RStartPos: 60,
  circleYDif: 100,
  smallerRounded: 15,
};

export enum ChartLabel {
  Calories = "ENERC_KCAL",
  Protein = "PROCNT",
  Fat = "FAT",
  Saturates = "FASAT",
  Sugars = "SUGAR",
  Salt = "NA",
}

export enum Colour {
  Green = 1,
  Yellow = 2,
  Red = 3,
}
