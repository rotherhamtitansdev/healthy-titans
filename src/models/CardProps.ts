export interface CardProps {
  name: string;
  additionalStyling?: string;
  onClick?: () => void;
}
