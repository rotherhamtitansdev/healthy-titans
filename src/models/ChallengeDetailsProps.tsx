export interface ChallengeDetailsProps {
  name: string;
  description: string;
  img: string;
  rules: string[];
  equipment: string[];
  howToPlay?: string[];
}
