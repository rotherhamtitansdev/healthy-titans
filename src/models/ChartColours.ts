export interface ChartColours {
  name: string;
  primary: string;
  secondary: string;
  status: number;
  position: number;
}
