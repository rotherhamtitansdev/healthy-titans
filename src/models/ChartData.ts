export interface ChartData {
  Calories: number;
  Protein: number;
  Fat: number;
  Saturates: number;
  Sugars: number;
  Salt: number;
}