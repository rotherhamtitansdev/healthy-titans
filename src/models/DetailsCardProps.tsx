export interface ListProps {
title: string;
items: string[];
}
export interface DetailsCardProps {
  name?: string;
  img?: string;
  description?: string;
  list?: ListProps;
  hideTitle?: boolean;
  additionalStyling?: string;
  firebaseName?: string;
}
