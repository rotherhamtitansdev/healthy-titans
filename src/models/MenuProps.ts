import React from "react";
import { HeaderProps } from "./HeaderProps";
import { MenuCardProps } from "./MenuCardProps";
import { MenuTitleProps } from "./MenuTitleProps";

export interface MenuProps {
  header?: HeaderProps;
  cards: MenuCardProps[];
  title?: MenuTitleProps;
  children?: React.ReactNode;
}
