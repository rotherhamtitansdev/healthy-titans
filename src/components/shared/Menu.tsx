import React from "react";
import { MenuProps } from "../../models/MenuProps";
import AppHeader from "../app_header/AppHeader";
import MenuHeader from "../app_header/header/MenuHeader";
import MenuCard from "./MenuCard";
import MenuTitle from "./MenuTitle";

const Menu = (props: MenuProps) => {
  const smallMenu: boolean = props.cards.length < 3;

  function getStylingForCard(index: number) {
    if (!smallMenu) {
      let styling = "lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem]";
      if ((index === props.cards.length - 1) && props.cards.length % 2 !== 0) {
        styling += " col-span-2 mx-[25%] lg:col-span-1 lg:mx-0";
      }
      return styling;
    }
    return "";
  }

  function getStyling() {
    if (smallMenu) {
      return "px-5 py-5 grid grid-cols-1 lg:grid-cols-2 lg:gap-x-10 gap-y-6";
    }
    return "grid grid-cols-2 xs:p-4 lg:grid-cols-auto lg:px-20 2xl:px-36 lg:py-10 gap-[0.5rem] md:gap-y-[1rem] lg:gap-y-[3rem]";
  }

  const cardList = props.cards.map((item, index) => (
    <MenuCard
      key={item.key}
      name={item.name}
      img={item.img}
      path={item.path}
      externalPath={item.externalPath}
      additionalStyling={getStylingForCard(index)}
      firebaseName={item.firebaseName}
    />
  ));

  return (
    <>
      {props.header ? (
        <AppHeader>
          <MenuHeader title={props.header.title} body={props.header.body} />
        </AppHeader>
      ) : null}
      <div className="p-5 md:p-10 pb-20">
        {props.title ? (
          <MenuTitle
            title={props.title.title}
            subtitle={props.title.subtitle}
          />
        ) : null}
        <div className={getStyling()}>
          {cardList}
          {!smallMenu ? (
            <div className="grid col-span-2 lg:col-span-3 ">
              {props.children}
            </div>
          ) : null}
        </div>
      </div>
    </>
  );
};
export default Menu;
