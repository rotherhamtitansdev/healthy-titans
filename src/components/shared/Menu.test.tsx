import React from "react";
import { screen, render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Menu from "./Menu";
import { MenuProps } from "../../models/MenuProps";

describe("Menu Component", () => {
  test("if component renders header, card, title, but not children (less than 3 cards)", () => {
    const mockProps: MenuProps = {
      header: { title: "Test First Title", body: "Test Body Title" },
      cards: [{ key: 1, name: "Test Card Name 1", path: "https://www.bbc.co.uk/" }],
      title: { title: "Test Second Title" }
    };
    const { asFragment } = render(
      <Menu cards={mockProps.cards} header={mockProps.header} title={mockProps.title}>
        <h1>Hello</h1>
      </Menu>,
      { wrapper: MemoryRouter }
    );
    expect(screen.getByRole("heading", {name: "Test First Title"})).toBeVisible();
    expect(screen.getByText("Test Second Title")).toBeVisible();
    expect(screen.queryByText("Hello")).not.toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test("if component renders header, card, title, and children (3+ cards)", () => {
    const mockProps: MenuProps = {
      header: { title: "Test First Title", body: "Test Body Title" },
      cards: [{ key: 1, name: "Test Card Name 1", path: "https://www.bbc.co.uk/" },{ key: 2, name: "Test Card Name 2", path: "https://www.google.co.uk/" },{ key: 3, name: "Test Card Name 3", path: "https://www.yahoo.co.uk/" }],
      title: { title: "Test Second Title" }
    };
    const { asFragment } = render(
      <Menu cards={mockProps.cards} header={mockProps.header} title={mockProps.title}>
        <h1>Hello</h1>
      </Menu>,
      { wrapper: MemoryRouter }
    );
    expect(screen.getByRole("heading", {name: "Test First Title"})).toBeVisible();
    expect(screen.getByText("Test Second Title")).toBeVisible();
    expect(screen.getByText("Hello")).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });

  test("if component renders only cards, no header or title", () => {
    const mockProps: MenuProps = {
      cards: [{ key: 1, name: "Test Card Name 1", path: "https://www.bbc.co.uk/" },{ key: 2, name: "Test Card Name 2", path: "https://www.google.co.uk/" },{ key: 3, name: "Test Card Name 3", path: "https://www.yahoo.co.uk/" }],
    };
    const { asFragment } = render(
      <Menu cards={mockProps.cards} />,
      { wrapper: MemoryRouter }
    );
    
    expect(screen.getByText("Test Card Name 1")).toBeVisible();
    expect(screen.getByText("Test Card Name 2")).toBeVisible();
    expect(screen.getByText("Test Card Name 3")).toBeVisible();
    expect(screen.queryByTestId("header-title")).not.toBeInTheDocument();
    expect(screen.queryByTestId("menutitle-title")).not.toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test("if conditional styling is applied, less than 3 cards", () => {
    const mockProps: MenuProps = {
      header: { title: "Test First Title", body: "Test Body Title" },
      cards: [{ key: 1, name: "Test Card Name 1", path: "https://www.bbc.co.uk/" }],
      title: { title: "Test Second Title" }
    };
    const { asFragment } = render(
      <Menu cards={mockProps.cards} header={mockProps.header} title={mockProps.title}>
        <h1>Hello</h1>
      </Menu>,
      { wrapper: MemoryRouter }
    );
    
    expect(screen.getByTestId("menu-styling")).toHaveClass("px-5 py-5 grid grid-cols-1 lg:grid-cols-2 lg:gap-x-10 gap-y-6")
    expect(asFragment()).toMatchSnapshot();
  });

  test("if conditional styling is applied, 3+ cards", () => {
    const mockProps: MenuProps = {
      header: { title: "Test First Title", body: "Test Body Title" },
      cards: [{ key: 1, name: "Test Card Name 1", path: "https://www.bbc.co.uk/" },{ key: 2, name: "Test Card Name 2", path: "https://www.google.co.uk/" },{ key: 3, name: "Test Card Name 3", path: "https://www.yahoo.co.uk/" }],
      title: { title: "Test Second Title" }
    };
    const { asFragment } = render(
      <Menu cards={mockProps.cards} header={mockProps.header} title={mockProps.title}>
        <h1>Hello</h1>
      </Menu>,
      { wrapper: MemoryRouter }
    );
    
    expect(screen.getByTestId("menu-styling")).toHaveClass("grid grid-cols-2 xs:p-4 lg:grid-cols-auto lg:px-14 xl:px-20 2xl:px-30 3xl:px-36 lg:py-10 gap-[0.5rem] md:gap-y-[1rem] lg:gap-y-[3rem]")
    expect(screen.getByTestId(mockProps.cards[0].name)).toHaveClass("lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem]")
    expect(screen.getByTestId(mockProps.cards[1].name)).toHaveClass("lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem]")
    expect(screen.getByTestId(mockProps.cards[2].name)).toHaveClass("lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem] col-span-2 mx-[25%] lg:col-span-1 lg:mx-0")
    expect(asFragment()).toMatchSnapshot();
  });
});
