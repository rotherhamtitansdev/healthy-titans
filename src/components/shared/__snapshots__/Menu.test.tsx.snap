// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component if component renders header, card, title, and children (3+ cards) 1`] = `
<DocumentFragment>
  <div>
    <div
      class="bg-mobileWave sm:bg-wave bg-no-repeat pb-2 bg-span"
    >
      <div
        class="flex flex-wrap h-[22rem]"
      >
        <div
          class="mx-auto pt-[3rem] md:pt-[2rem]"
        >
          <h1
            class="text-titansDarkBlue font-bold text-center sm:mt-[1rem] text-3xl sm:text-5xl lg:text-6xl mt-8 pb-4"
            data-testid="header-title"
          >
            Test First Title
          </h1>
          <div
            class="font-medium text-homepageHeaderSubTitle text-sm md:text-base lg:text-xl text-center w-2/3 lg:w-[45rem] mx-auto"
          >
            Test Body Title
          </div>
        </div>
      </div>
    </div>
    <div
      class="p-5 md:p-10 pb-20"
    >
      <div
        class="md:px-5"
      >
        <div
          class="font-semibold text-[#243A73] text-2xl md:text-3xl lg:text-4xl py-5"
          data-testid="menutitle-title"
        >
          Test Second Title
        </div>
        <div
          class="text-[#243A73] flex font-medium md:text-xl lg:text-xl pb-5"
          data-testid="breadcrumbs"
        />
      </div>
      <div
        class="grid grid-cols-2 xs:p-4 lg:grid-cols-auto lg:px-14 xl:px-20 2xl:px-30 3xl:px-36 lg:py-10 gap-[0.5rem] md:gap-y-[1rem] lg:gap-y-[3rem]"
        data-testid="menu-styling"
      >
        <div
          class="bg-white table align-middle rounded-3xl shadow-lg lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem] hover:cursor-grab hover:-translate-y-0.5 hover:scale-105 duration-300"
          data-testid="Test Card Name 1"
          role="button"
          tabindex="0"
        >
          <div
            class="py-[1.75rem] md:py-[3.625rem] w-full h-full flex-col flex"
          >
            <div
              class="w-full h-full flex justify-center items-center md:-mt-0 md:-mt-8"
            >
              <span
                class=""
                style="width: 100px; height: 100px; border-radius: 100%; border: 2px solid; border-top-color: #DC476D; border-bottom-color: transparent; border-left-color: #DC476D; border-right-color: #DC476D; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
              />
            </div>
            <div
              class="md:text-xl text-homepageHeaderText font-semibold text-center mt-auto"
            >
              Test Card Name 1
            </div>
          </div>
        </div>
        <div
          class="bg-white table align-middle rounded-3xl shadow-lg lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem] hover:cursor-grab hover:-translate-y-0.5 hover:scale-105 duration-300"
          data-testid="Test Card Name 2"
          role="button"
          tabindex="0"
        >
          <div
            class="py-[1.75rem] md:py-[3.625rem] w-full h-full flex-col flex"
          >
            <div
              class="w-full h-full flex justify-center items-center md:-mt-0 md:-mt-8"
            >
              <span
                class=""
                style="width: 100px; height: 100px; border-radius: 100%; border: 2px solid; border-top-color: #DC476D; border-bottom-color: transparent; border-left-color: #DC476D; border-right-color: #DC476D; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
              />
            </div>
            <div
              class="md:text-xl text-homepageHeaderText font-semibold text-center mt-auto"
            >
              Test Card Name 2
            </div>
          </div>
        </div>
        <div
          class="bg-white table align-middle rounded-3xl shadow-lg lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem] col-span-2 mx-[25%] lg:col-span-1 lg:mx-0 hover:cursor-grab hover:-translate-y-0.5 hover:scale-105 duration-300"
          data-testid="Test Card Name 3"
          role="button"
          tabindex="0"
        >
          <div
            class="py-[1.75rem] md:py-[3.625rem] w-full h-full flex-col flex"
          >
            <div
              class="w-full h-full flex justify-center items-center md:-mt-0 md:-mt-8"
            >
              <span
                class=""
                style="width: 100px; height: 100px; border-radius: 100%; border: 2px solid; border-top-color: #DC476D; border-bottom-color: transparent; border-left-color: #DC476D; border-right-color: #DC476D; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
              />
            </div>
            <div
              class="md:text-xl text-homepageHeaderText font-semibold text-center mt-auto"
            >
              Test Card Name 3
            </div>
          </div>
        </div>
        <div
          class="grid col-span-2 lg:col-span-3 "
        >
          <h1>
            Hello
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Menu Component if component renders header, card, title, but not children (less than 3 cards) 1`] = `
<DocumentFragment>
  <div>
    <div
      class="bg-mobileWave sm:bg-wave bg-no-repeat pb-2 bg-span"
    >
      <div
        class="flex flex-wrap h-[22rem]"
      >
        <div
          class="mx-auto pt-[3rem] md:pt-[2rem]"
        >
          <h1
            class="text-titansDarkBlue font-bold text-center sm:mt-[1rem] text-3xl sm:text-5xl lg:text-6xl mt-8 pb-4"
            data-testid="header-title"
          >
            Test First Title
          </h1>
          <div
            class="font-medium text-homepageHeaderSubTitle text-sm md:text-base lg:text-xl text-center w-2/3 lg:w-[45rem] mx-auto"
          >
            Test Body Title
          </div>
        </div>
      </div>
    </div>
    <div
      class="p-5 md:p-10 pb-20"
    >
      <div
        class="md:px-5"
      >
        <div
          class="font-semibold text-[#243A73] text-2xl md:text-3xl lg:text-4xl py-5"
          data-testid="menutitle-title"
        >
          Test Second Title
        </div>
        <div
          class="text-[#243A73] flex font-medium md:text-xl lg:text-xl pb-5"
          data-testid="breadcrumbs"
        />
      </div>
      <div
        class="px-5 py-5 grid grid-cols-1 lg:grid-cols-2 lg:gap-x-10 gap-y-6"
        data-testid="menu-styling"
      >
        <div
          class="bg-white table align-middle rounded-3xl shadow-lg  hover:cursor-grab hover:-translate-y-0.5 hover:scale-105 duration-300"
          data-testid="Test Card Name 1"
          role="button"
          tabindex="0"
        >
          <div
            class="py-[1.75rem] md:py-[3.625rem] w-full h-full flex-col flex"
          >
            <div
              class="w-full h-full flex justify-center items-center md:-mt-0 md:-mt-8"
            >
              <span
                class=""
                style="width: 100px; height: 100px; border-radius: 100%; border: 2px solid; border-top-color: #DC476D; border-bottom-color: transparent; border-left-color: #DC476D; border-right-color: #DC476D; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
              />
            </div>
            <div
              class="md:text-xl text-homepageHeaderText font-semibold text-center mt-auto"
            >
              Test Card Name 1
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Menu Component if component renders only cards, no header or title 1`] = `
<DocumentFragment>
  <div>
    <div
      class="p-5 md:p-10 pb-20"
    >
      <div
        class="grid grid-cols-2 xs:p-4 lg:grid-cols-auto lg:px-14 xl:px-20 2xl:px-30 3xl:px-36 lg:py-10 gap-[0.5rem] md:gap-y-[1rem] lg:gap-y-[3rem]"
        data-testid="menu-styling"
      >
        <div
          class="bg-white table align-middle rounded-3xl shadow-lg lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem] hover:cursor-grab hover:-translate-y-0.5 hover:scale-105 duration-300"
          data-testid="Test Card Name 1"
          role="button"
          tabindex="0"
        >
          <div
            class="py-[1.75rem] md:py-[3.625rem] w-full h-full flex-col flex"
          >
            <div
              class="w-full h-full flex justify-center items-center md:-mt-0 md:-mt-8"
            >
              <span
                class=""
                style="width: 100px; height: 100px; border-radius: 100%; border: 2px solid; border-top-color: #DC476D; border-bottom-color: transparent; border-left-color: #DC476D; border-right-color: #DC476D; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
              />
            </div>
            <div
              class="md:text-xl text-homepageHeaderText font-semibold text-center mt-auto"
            >
              Test Card Name 1
            </div>
          </div>
        </div>
        <div
          class="bg-white table align-middle rounded-3xl shadow-lg lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem] hover:cursor-grab hover:-translate-y-0.5 hover:scale-105 duration-300"
          data-testid="Test Card Name 2"
          role="button"
          tabindex="0"
        >
          <div
            class="py-[1.75rem] md:py-[3.625rem] w-full h-full flex-col flex"
          >
            <div
              class="w-full h-full flex justify-center items-center md:-mt-0 md:-mt-8"
            >
              <span
                class=""
                style="width: 100px; height: 100px; border-radius: 100%; border: 2px solid; border-top-color: #DC476D; border-bottom-color: transparent; border-left-color: #DC476D; border-right-color: #DC476D; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
              />
            </div>
            <div
              class="md:text-xl text-homepageHeaderText font-semibold text-center mt-auto"
            >
              Test Card Name 2
            </div>
          </div>
        </div>
        <div
          class="bg-white table align-middle rounded-3xl shadow-lg lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem] col-span-2 mx-[25%] lg:col-span-1 lg:mx-0 hover:cursor-grab hover:-translate-y-0.5 hover:scale-105 duration-300"
          data-testid="Test Card Name 3"
          role="button"
          tabindex="0"
        >
          <div
            class="py-[1.75rem] md:py-[3.625rem] w-full h-full flex-col flex"
          >
            <div
              class="w-full h-full flex justify-center items-center md:-mt-0 md:-mt-8"
            >
              <span
                class=""
                style="width: 100px; height: 100px; border-radius: 100%; border: 2px solid; border-top-color: #DC476D; border-bottom-color: transparent; border-left-color: #DC476D; border-right-color: #DC476D; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
              />
            </div>
            <div
              class="md:text-xl text-homepageHeaderText font-semibold text-center mt-auto"
            >
              Test Card Name 3
            </div>
          </div>
        </div>
        <div
          class="grid col-span-2 lg:col-span-3 "
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Menu Component if conditional styling is applied, 3+ cards 1`] = `
<DocumentFragment>
  <div>
    <div
      class="bg-mobileWave sm:bg-wave bg-no-repeat pb-2 bg-span"
    >
      <div
        class="flex flex-wrap h-[22rem]"
      >
        <div
          class="mx-auto pt-[3rem] md:pt-[2rem]"
        >
          <h1
            class="text-titansDarkBlue font-bold text-center sm:mt-[1rem] text-3xl sm:text-5xl lg:text-6xl mt-8 pb-4"
            data-testid="header-title"
          >
            Test First Title
          </h1>
          <div
            class="font-medium text-homepageHeaderSubTitle text-sm md:text-base lg:text-xl text-center w-2/3 lg:w-[45rem] mx-auto"
          >
            Test Body Title
          </div>
        </div>
      </div>
    </div>
    <div
      class="p-5 md:p-10 pb-20"
    >
      <div
        class="md:px-5"
      >
        <div
          class="font-semibold text-[#243A73] text-2xl md:text-3xl lg:text-4xl py-5"
          data-testid="menutitle-title"
        >
          Test Second Title
        </div>
        <div
          class="text-[#243A73] flex font-medium md:text-xl lg:text-xl pb-5"
          data-testid="breadcrumbs"
        />
      </div>
      <div
        class="grid grid-cols-2 xs:p-4 lg:grid-cols-auto lg:px-14 xl:px-20 2xl:px-30 3xl:px-36 lg:py-10 gap-[0.5rem] md:gap-y-[1rem] lg:gap-y-[3rem]"
        data-testid="menu-styling"
      >
        <div
          class="bg-white table align-middle rounded-3xl shadow-lg lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem] hover:cursor-grab hover:-translate-y-0.5 hover:scale-105 duration-300"
          data-testid="Test Card Name 1"
          role="button"
          tabindex="0"
        >
          <div
            class="py-[1.75rem] md:py-[3.625rem] w-full h-full flex-col flex"
          >
            <div
              class="w-full h-full flex justify-center items-center md:-mt-0 md:-mt-8"
            >
              <span
                class=""
                style="width: 100px; height: 100px; border-radius: 100%; border: 2px solid; border-top-color: #DC476D; border-bottom-color: transparent; border-left-color: #DC476D; border-right-color: #DC476D; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
              />
            </div>
            <div
              class="md:text-xl text-homepageHeaderText font-semibold text-center mt-auto"
            >
              Test Card Name 1
            </div>
          </div>
        </div>
        <div
          class="bg-white table align-middle rounded-3xl shadow-lg lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem] hover:cursor-grab hover:-translate-y-0.5 hover:scale-105 duration-300"
          data-testid="Test Card Name 2"
          role="button"
          tabindex="0"
        >
          <div
            class="py-[1.75rem] md:py-[3.625rem] w-full h-full flex-col flex"
          >
            <div
              class="w-full h-full flex justify-center items-center md:-mt-0 md:-mt-8"
            >
              <span
                class=""
                style="width: 100px; height: 100px; border-radius: 100%; border: 2px solid; border-top-color: #DC476D; border-bottom-color: transparent; border-left-color: #DC476D; border-right-color: #DC476D; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
              />
            </div>
            <div
              class="md:text-xl text-homepageHeaderText font-semibold text-center mt-auto"
            >
              Test Card Name 2
            </div>
          </div>
        </div>
        <div
          class="bg-white table align-middle rounded-3xl shadow-lg lg:w-[17rem] xl:w-[21rem] xl:h-[21rem] 2xl:w-[25rem] 2xl:h-[25rem] col-span-2 mx-[25%] lg:col-span-1 lg:mx-0 hover:cursor-grab hover:-translate-y-0.5 hover:scale-105 duration-300"
          data-testid="Test Card Name 3"
          role="button"
          tabindex="0"
        >
          <div
            class="py-[1.75rem] md:py-[3.625rem] w-full h-full flex-col flex"
          >
            <div
              class="w-full h-full flex justify-center items-center md:-mt-0 md:-mt-8"
            >
              <span
                class=""
                style="width: 100px; height: 100px; border-radius: 100%; border: 2px solid; border-top-color: #DC476D; border-bottom-color: transparent; border-left-color: #DC476D; border-right-color: #DC476D; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
              />
            </div>
            <div
              class="md:text-xl text-homepageHeaderText font-semibold text-center mt-auto"
            >
              Test Card Name 3
            </div>
          </div>
        </div>
        <div
          class="grid col-span-2 lg:col-span-3 "
        >
          <h1>
            Hello
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Menu Component if conditional styling is applied, less than 3 cards 1`] = `
<DocumentFragment>
  <div>
    <div
      class="bg-mobileWave sm:bg-wave bg-no-repeat pb-2 bg-span"
    >
      <div
        class="flex flex-wrap h-[22rem]"
      >
        <div
          class="mx-auto pt-[3rem] md:pt-[2rem]"
        >
          <h1
            class="text-titansDarkBlue font-bold text-center sm:mt-[1rem] text-3xl sm:text-5xl lg:text-6xl mt-8 pb-4"
            data-testid="header-title"
          >
            Test First Title
          </h1>
          <div
            class="font-medium text-homepageHeaderSubTitle text-sm md:text-base lg:text-xl text-center w-2/3 lg:w-[45rem] mx-auto"
          >
            Test Body Title
          </div>
        </div>
      </div>
    </div>
    <div
      class="p-5 md:p-10 pb-20"
    >
      <div
        class="md:px-5"
      >
        <div
          class="font-semibold text-[#243A73] text-2xl md:text-3xl lg:text-4xl py-5"
          data-testid="menutitle-title"
        >
          Test Second Title
        </div>
        <div
          class="text-[#243A73] flex font-medium md:text-xl lg:text-xl pb-5"
          data-testid="breadcrumbs"
        />
      </div>
      <div
        class="px-5 py-5 grid grid-cols-1 lg:grid-cols-2 lg:gap-x-10 gap-y-6"
        data-testid="menu-styling"
      >
        <div
          class="bg-white table align-middle rounded-3xl shadow-lg  hover:cursor-grab hover:-translate-y-0.5 hover:scale-105 duration-300"
          data-testid="Test Card Name 1"
          role="button"
          tabindex="0"
        >
          <div
            class="py-[1.75rem] md:py-[3.625rem] w-full h-full flex-col flex"
          >
            <div
              class="w-full h-full flex justify-center items-center md:-mt-0 md:-mt-8"
            >
              <span
                class=""
                style="width: 100px; height: 100px; border-radius: 100%; border: 2px solid; border-top-color: #DC476D; border-bottom-color: transparent; border-left-color: #DC476D; border-right-color: #DC476D; display: inline-block; animation: react-spinners-ClipLoader-clip 0.75s 0s infinite linear; animation-fill-mode: both;"
              />
            </div>
            <div
              class="md:text-xl text-homepageHeaderText font-semibold text-center mt-auto"
            >
              Test Card Name 1
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
